packages/datetime/src/datePicker.tsx

```diff
@@ -269,11 +301,62 @@ export class DatePicker extends AbstractPureComponent2<IDatePickerProps, IDatePi
         );
     }
 
+    private maybeRenderShortcuts() {
+        let { shortcuts } = this.props;
+        if (shortcuts == null || shortcuts === false) {
+            return null;
+        }
+
+        const { selectedShortcutIndex } = this.state;
+        const { maxDate, minDate, timePrecision } = this.props;
+        // Reuse the existing date range shortcuts and only care about start date
+        const dateRangeShortcuts: IDateRangeShortcut[] | true = shortcuts === true ? true : shortcuts.map(shortcut => ({
+            ...shortcut,
+            dateRange: [shortcut.date, undefined],
+        }));
+        return [
+            <Shortcuts
+                key="shortcuts"
+                {...{ allowSingleDayRange: true, maxDate, minDate, shortcuts: dateRangeShortcuts, timePrecision, selectedShortcutIndex }}
+                onShortcutClick={this.handleShortcutClick}
+                useSingleDateLabels={true}
+            />,
+            <Divider key="div" />,
+        ];
+    }
+
     private handleDayClick = (day: Date, modifiers: DayModifiers, e: React.MouseEvent<HTMLDivElement>) => {
         Utils.safeInvoke(this.props.dayPickerProps.onDayClick, day, modifiers, e);
         if (modifiers.disabled) {
             return;
         }
+
+        this.updateDay(day);
+
+        // allow toggling selected date by clicking it again (if prop enabled)
+        const newValue =
+            this.props.canClearSelection && modifiers.selected ? null : DateUtils.getDateTime(day, this.state.value);
+        this.updateValue(newValue, true);
+    };
+
+    private handleShortcutClick = (shortcut: IDateRangeShortcut, selectedShortcutIndex: number) => {
+        const { onShortcutChange } = this.props;
+        const { dateRange, includeTime } = shortcut;
+        const newDate = dateRange[0];
+        const newValue = includeTime ? newDate : DateUtils.getDateTime(newDate, this.state.value);
+
+        this.updateDay(newDate);
+        this.updateValue(newValue, true);
+
+        if (this.props.selectedShortcutIndex === undefined) {
+            this.setState({ selectedShortcutIndex });
+        }
+
+        const datePickerShortcut = { ...shortcut, date: shortcut.dateRange[0] };
+        Utils.safeInvoke(onShortcutChange, datePickerShortcut, selectedShortcutIndex);
+    };
+
+    private updateDay = (day: Date) => {
```

Comment:
pulled out this common logic for when clicking a shortcut vs. a date on the calendar